src/main/java/org/codehaus/mojo/license/AddThirdPartyMojo.java

```diff
@@ -469,7 +469,7 @@ private void writeMissingFile()
         }
     }
 
-    void initFromMojo( AggregatorAddThirdPartyMojo mojo, MavenProject mavenProject,
+    void initFromMojo( AbstractAddThirdPartyMojo mojo, MavenProject mavenProject,
```

Comment:
Is this change necessary to fix the issue? IIRC, AggregatorAddThirdPartyMojo uses (and instantiates) AbstractAddThirdPartyMojo to do the job for the submodules. So this method makes sense only with AggregatorAddThirdPartyMojo arg. I might be wrong, I am not an expert in these mojos.

Reply:
No, it is not needed. But
> AggregatorAddThirdPartyMojo uses (and instantiates) AbstractAddThirdPartyMojo to do the job for the submodules

the AggregatorAddThirdPartyMojo uses and instantiates the AddThirdPartyMojo to do the job
```
AddThirdPartyMojo mojo = (AddThirdPartyMojo) getSession()
                    .lookup( AddThirdPartyMojo.ROLE, addThirdPartyRoleHint );
mojo.initFromMojo( this, reactorProject, new ArrayList<>( this.reactorProjects ) );
``` 
and AddThirdPartyMojo takes exactly the AggregatorAddThirdPartyMojo. This sound as a circulas dependencies. Since all the task is performed for each module in the reactor it's like run add-third-party goal directly. Anyway I can revert if you would keep this hard couple wiring.